proknow/Patients/Patients.py

```diff
@@ -636,6 +636,48 @@ def find_entities(self, predicate=None, **props):
                                 entities.append(entity_m)
         return entities
 
+    def find_sros(self, predicate=None, **props):
+        """Finds the sros for the patient matching the input paramters.
+
+        Note:
+            For more information on how to use this method, see :ref:`find-methods`.
+
+        Parameters:
+            predicate (func): A function that is passed an entity as input and which should return
+                a bool indicating whether the entity is a match.
+            **props: A dictionary of keyword arguments that may include any entity attribute to
+                match. These arguments are considered in turn to find matching entities.
+
+        Returns:
+            list: A list of matching :class:`proknow.Patients.SroSummary` objects.
+
+        Example:
+            Use this example to find the patient entities matching the predicate function (returns
+            all sros)::
+
+                from proknow import ProKnow
+
+                pk = ProKnow('https://example.proknow.com', credentials_file="./credentials.json")
+                patients = pk.patients.lookup("Clinical", ["HNC-0522c0009"])
+                patient = patients[0].get()
+                entities = patient.find_sros(lambda sro: True)
+        """
+        if predicate is None and len(props) == 0:
+            return []
+
+        sros = []
+        for study in self.studies:
+            for sro in study.sros:
+                match = True
+                for key in props:
+                    if sro.data[key] != props[key]:
```

Comment:
If `key` doesn't exist in `sro.data`, this will error out. If that's not possible, disregard. Otherwise might be good to check if it's in there with `sro.data.get(key)`

Reply:
I think we want it to error out in this case because it would indicate to the user that that property isn't available in the object and they shouldn't include that key in their search.